packages/msw-addon/src/loader.ts

```diff
@@ -0,0 +1,45 @@
+import type { RequestHandler } from 'msw'
+import type { Context } from './decorator'
+import { api } from '@build-time/initialize'
+
+export const mswLoader = async (context: Context) => {
+  const {
+    parameters: { msw },
+  } = context
+
+  if (api) {
+    api.resetHandlers()
+
+    if (msw) {
+      if (Array.isArray(msw) && msw.length > 0) {
```

Comment:
My understanding is we can just get rid of the decorator entirely. That it was the original implementation but has timing issues that are solved by the loader.

Reply:
@yannbf, can you please confirm that? 

If we can drop `mswDecorator`, we can simplify our addon _tremendously_. Now is also the right time to drop it, since the MSW 2.0 introduction is a breaking change anyway. 

I'd also like to discuss the general DX with you. I feel that we can move the `initialize` call to the MSW loader and have a better end developer-experience for everyone. My assumption is that we are asking the users to call `initialize` in Storybook's config to kick off MSW _earlier_. Is that the case?

Reply:
Yup I had plans to remove the mswdecorator. The problem is that it makes it much more difficult for people who reuse stories in tests (especially because portable stories do not support loaders), and this might be a big change for them, and thus requires a proper migration guide.

Regarding initialize, we could provide the code internally but then we remove the customizing aspect of it, which is needed in many cases. We could also make it optional, and call the internal code only if initialize is not called by the user. The side effect to this, again, is that it will break the story reuse scenario (because portable stories won't access such presets)

Reply:
Got it. Thanks for providing the much needed context here!

> Regarding initialize, we could provide the code internally but then we remove the customizing aspect of it, which is needed in many cases.

Can't we provide the customization on the loader level though?

```js
export function mswLoader(options) {
  return theActualLoader
}

{
  loaders: [mswLoader({ onUnhandledRequest: 'warn' })
}
```